File: fixtures/sample_repo/backend/dispatch.py
from backend.services.notification_service import send_email, send_sms

HANDLERS = {
    "email": send_email,
    "sms": send_sms,
}


def dispatch(kind, user, message):
    # Dynamic/ambiguous call site: static analysis cannot know which handler
    # runs without evaluating `kind` at runtime. Resolvers should emit this
    # edge as low-confidence, fanning out to every value in HANDLERS rather
    # than guessing a single target.
    handler = HANDLERS[kind]
    return handler(user, message)

File: fixtures/sample_repo/frontend/utils/dispatch.js
function onClick(event) {
  return `clicked ${event}`;
}

function onHover(event) {
  return `hovered ${event}`;
}

const handlers = {
  click: onClick,
  hover: onHover,
};

export function dispatch(name, event) {
  // Dynamic/ambiguous call site, mirrors backend/dispatch.py: the target
  // depends on the runtime value of `name`, so this must resolve to a
  // low-confidence fan-out edge across every value in `handlers`.
  return handlers[name](event);
}
